### PR TITLE
fix(anvil): for deposit transactions apply tx.mint value in tx validation 

### PR DESCRIPTION
**Pull request by jakim929**
## Motivation

There was a bug in how deposit_tx are validated in the tx pool. The actual logic in revm for how the tx is applied is correct, this just pertains to how anvil pre-validates the tx.

### repro

send a deposit tx using `eth_sendRawTransaction` with a `from` that does not have any eth balance, but with a `mint` to cover the value. It will fail the check `account.balance > gas * price + value`, because the sender doesn't have any balance to cover the gas costs / value.

As stated [here in the OP stack specs](https://specs.optimism.io/protocol/deposits.html#execution), we need to increase the balance of the account before running balance validation. It should also skip gas cost checks since gas is prepaid on the L1.

## Solution

if the tx is a deposit tx, simply check that `tx.value <= existing account value + deposited value` and don't check gas costs.

### CODE REVIEW

File: crates/anvil/tests/it/optimism.rs [hunk 1]
```diff
@@ -165,52 +164,60 @@ async fn test_deposit_transaction_hash_matches_sepolia() {
...
-        .await
-        .unwrap();
+#[tokio::test(flavor = "multi_thread")]
+async fn test_deposit_tx_checks_sufficient_funds_after_applying_deposited_value() {
+    // enable the Optimism flag
+    let (_api, handle) =
+        spawn(NodeConfig::test().with_optimism(true).with_hardfork(Some(Hardfork::Paris))).await;
 
-    // mine block
-    api.evm_mine(None).await.unwrap();
```
Comment (jakim929) [inline crates/anvil/tests/it/optimism.rs:209]: there was a small race condition here.

If the deposit-tx is sent then automined, it isn't evicted from the tx pool fast enough (with `Pool.prune_markers()`) so the same tx is included in the next block as well.

File: crates/anvil/tests/it/optimism.rs [hunk 2]
```diff
@@ -165,52 +164,60 @@ async fn test_deposit_transaction_hash_matches_sepolia() {
...
-        .with_nonce(0)
-        .with_from(sender_addr)
-        .with_to(deposit_tx_from)
-        .with_value(U256::from(1e18))
-        .with_gas_limit(21_000)
-        .with_max_fee_per_gas(20_000_000_000)
-        .with_max_priority_fee_per_gas(1_000_000_000);
-
-    provider
-        .send_transaction(WithOtherFields::new(fund_account_tx))
```
Comment (jakim929) [inline crates/anvil/tests/it/optimism.rs:190]: with the fix, this now passes without explicitly funding the account

File: crates/anvil/src/eth/backend/mem/mod.rs [hunk 3]
```diff
@@ -2573,16 +2573,31 @@ impl TransactionValidator for Backend {
...
-            tx.hash());
-            InvalidTransactionError::InsufficientFunds
-        })?;
-        if account.balance < U256::from(req_funds) {
-            warn!(target: "backend", "[{:?}] insufficient allowance={}, required={} account={:?}", tx.hash(), account.balance, req_funds, *pending.sender());
-            return Err(InvalidTransactionError::InsufficientFunds);
+
+        match &tx.transaction {
+            TypedTransaction::Deposit(deposit_tx) => {
+                // Deposit transactions
```
Comment (klkvr) [inline crates/anvil/src/eth/backend/mem/mod.rs:2579]: let's link OP doc here

**Review by klkvr — APPROVED** [review]
lgtm, just a doc nit

File: crates/anvil/src/eth/backend/mem/mod.rs (hunk 3 above)
Comment (jakim929) [reply crates/anvil/src/eth/backend/mem/mod.rs:2579]: good point, added!

**Pull request merged**